codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionTest.java

```diff
@@ -368,6 +368,21 @@ public void clientLocalIncrementAndGetStreamShouldRespectOverflow() throws Http2
         incrementAndGetStreamShouldRespectOverflow(client.local(), MAX_VALUE);
     }
 
+    @Test
+    public void incrementAndGetStreamShouldAlwaysIncrement() {
+        final int streamId = client.local().lastStreamKnownByPeer() + 1;
+
+        assertThrows(Http2Exception.class, new Executable() {
+            @Override
+            public void execute() throws Throwable {
+                client.local().createStream(streamId, true);
+            }
+        });
+
+        int nextStreamID = client.local().incrementAndGetNextStreamId();
+        assertEquals(streamId, nextStreamID - 3);
```

Comment:
It's hard to see how the change helps with the current test. Will be awesome if you can add a full integration test similar to the reproducer provided in #12065.

Reply:
Spent most of my time today trying to create a new test in `codec-http2/src/test/java/io/netty/handler/codec/http2/Http2ConnectionRoundtripTest.java` but having a hard time working with the api to recreate what the outcomes of https://github.com/netty/netty/issues/12065 are. Got any points/starting off points?

Apologies, first time working in this repo.

Reply:
seems like you did not push it yet ? Please do and I will have a look.

Reply:
let me merge this one first and I will work on a more complex test-case. The fix is correct